persistent/Database/Persist/Quasi/Internal.hs

```diff
@@ -98,8 +99,16 @@ parseFieldType t0 =
             PSDone -> PSSuccess (front []) t
             -- _ ->
 
+newtype ToFKName = ToFKName
+  { unToFKName :: Text -> Text -> Text
+  }
+
 data PersistSettings = PersistSettings
     { psToDBName :: !(Text -> Text)
+    , psToFKName :: !ToFKName
+    -- ^ Function used to generate the FK name. Default value: @mappend@
+    --
+    -- @since 2.12.1.2
```

Comment:
This PR is currently against `persistent-2.13` - are you planning on backporting it?

Note that the `PersistSettings` is currently hidden, so we'll want to export a setter `setPsToFKName :: ToFKName -> PersistSettings -> PersistSettings`.

Reply:
This is just a hangover from my original change request, I will update these `@since` declarations :+1: looks like I need to rethink how this is switched on as well as you say, as in `2.13` `PersistSettings` is now in an `Internal` module, so I will add something like `setPsToFKName`

Reply:
I've just pushed some more stuff up, so thats ready for a second look :+1: I will add some annotations to what I've done as there are some additional things in there to consider